lib/ansible/plugins/inventory/foreman.py

```diff
@@ -165,6 +165,8 @@ def _get_json(self, url, ignore_errors=None):
             # Set the cache if it is enabled or if the cache was refreshed
             if self.use_cache or self.get_option('cache'):
                 self._cache[self.cache_key][url] = results
+            else:
+                return results
```

Comment:
we should always update the _cache and return that, cache disabled just means we don't read from it initially, so i would remove the `if` instead

Reply:
+1. That does seem like a better way to solve this. If caching is not enabled the backing cache plugin is memory so always writing to it should work as expected.